src/Raven.Server/Documents/Indexes/VectorSearch/GenerateEmbeddings.cs

```diff
@@ -59,6 +59,76 @@ public static VectorValue FromText(ByteStringContext allocator, in VectorOptions
             Quantize(allocator, options.DestinationEmbeddingType, embedding.MemoryScope, embedding.Memory, embedding.UsedBytes);
     }
 
+    public static VectorValue FromValue(ByteStringContext allocator, IEmbeddingValue embedding, in VectorOptions options)
```

Comment:
This should probably not exists, `embedding` should be `Memory<byte>` or something like that

Reply:
Okay, maybe not perfect but that's the issue with that? Please check my answer about `IEmbeddingValue` interface above. This is actually "or something like that". It provides ability to have different types of embeddings that are somehow mixed:

- embedding from AI model
- embedding from AI model but quantized 
- embedding from local cache (attachment)